### PR TITLE
[5.1] OGM-982 - Upgrade to ORM 5.1

### PR DESCRIPTION
**Pull request by gunnarmorling**
Should be good to go now. I re-use the existing ORM 5.1.1-SNAPSHOT module ZIP now, so no need for the further "skinny module" approach. We only need to upgrade to 5.1.1.Final once out.

### CODE REVIEW

File: core/src/main/java/org/hibernate/ogm/persister/impl/OgmEntityPersister.java [hunk 1]
```diff
@@ -630,10 +630,10 @@ public Object initializeLazyProperty(String fieldName, Object entity, SessionImp
 			Object ce = getCacheAccessStrategy().get( session, cacheKey, session.getTimestamp() );
 			if ( ce != null ) {
 				CacheEntry cacheEntry = (CacheEntry) getCacheEntryStructure().destructure( ce, getFactory() );
-				if ( !cacheEntry.areLazyPropertiesUnfetched() ) {
-					// note early exit here:
-					return initializeLazyPropertiesFromCache( fieldName, entity, session, entry, cacheEntry );
-				}
+				final Object initializedValue = initializeLazyPropertiesFromCache( fieldName, entity, session, entry, cacheEntry );
+
+				// NOTE EARLY EXIT!!!
```
Comment (DavideD) [inline core/src/main/java/org/hibernate/ogm/persister/impl/OgmEntityPersister.java]: What's this comment is about?

Comment (DavideD) [inline core/src/main/java/org/hibernate/ogm/persister/impl/OgmEntityPersister.java]: Ok, I see that it was a pre-existing comment

Comment (gunnarmorling) [inline core/src/main/java/org/hibernate/ogm/persister/impl/OgmEntityPersister.java]: Yeah, I tried to deviate as less as possible from the original in ORM;
Dreaded C&P :/

Comment (DavideD) [inline core/src/main/java/org/hibernate/ogm/persister/impl/OgmEntityPersister.java]: No problem, I was just curious about it :)

**Comment by DavideD** [comment]
@gunnarmorling  It seems this PR is having problems with MongoDB can you have a look?

**Comment by gunnarmorling** [comment]
Yepp, on it already.

**Comment by gunnarmorling** [comment]
Jenkins, retest this please.

**Comment by gunnarmorling** [comment]
@DavideD Can you try and check whether it works for you locally? I cannot reproduce it, it seems. May be related to the snapshot dependency?

**Comment by DavideD** [comment]
ok

**Comment by DavideD** [comment]
Jenkins, retest this please

**Comment by gsmet** [comment]
I can reproduce it on my laptop. Will take a look.

**Comment by DavideD** [comment]
@gsmet I'm looking at it. It's related to the way embedded column names are returned. It is change in this version of ORM

**Comment by DavideD** [comment]
Ill send a PR shortly

**Comment by gsmet** [comment]
Ah, OK, wasn't sure you were able to reproduce it!

**Comment by gunnarmorling** [comment]
Cool, I suppose it may be related to that recent change from Guillaume in
ORM around the implicit naming strategy to which we need to adapt in OGM
now?

**Comment by DavideD** [comment]
Superseded by https://github.com/hibernate/hibernate-ogm/pull/726

**Comment by DavideD** [comment]
@gunnarmorling Yes, that was the problem. Plus a missing dependency in an integration test

**Pull request closed without merging**